crates/biome_css_analyze/src/lint/nursery/use_sorted_properties.rs

```diff
@@ -0,0 +1,387 @@
+    /// 1. Typography properties (font, color)
+    /// 1. Interaction properties (pointer-events, visibility)
+    /// 1. Background & border properties
+    /// 1. Transition & animation properties
+    /// 1. Nested rules, media queries & other at-rules
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// p { color: black; display: block; }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// p { display: block; color: black; }
+    /// ```
+    ///
+    pub UseSortedProperties {
+        version: "next",
+        name: "useSortedProperties",
+        language: "css",
+        recommended: true,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+pub struct UseSortedPropertiesState {
+    block: CssDeclarationOrRuleBlock,
+    original_properties: Vec<AnyCssDeclarationOrRule>,
+    sorted_properties: RecessOrderProperties,
+    is_unsafe_to_sort: bool,
+}
+
+impl Rule for UseSortedProperties {
+    type Query = Ast<CssDeclarationOrRuleBlock>;
+    type State = UseSortedPropertiesState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let query_result = ctx.query();
+
+        let original_properties = query_result
+            .items()
+            .into_iter()
+            .collect::<Vec<AnyCssDeclarationOrRule>>();
+        let is_unsafe_to_sort = contains_shorthand_after_longhand(&original_properties)
+            || contains_unknown_property(&original_properties);
+        let sorted_properties = RecessOrderProperties::new(&original_properties);
+
+        Some(UseSortedPropertiesState {
+            block: query_result.clone(),
+            original_properties,
+            sorted_properties,
+            is_unsafe_to_sort,
+        })
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        if state.is_unsafe_to_sort || state.sorted_properties.is_sorted() {
+            return None;
+        }
+
+        return Some(RuleDiagnostic::new(
+            rule_category!(),
+            state.block.range(),
+            markup! {
+                "Properties can be sorted."
+            },
+        ));
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<CssRuleAction> {
+        if state.is_unsafe_to_sort || state.sorted_properties.is_sorted() {
+            return None;
+        }
+
+        let mut mutation = ctx.root().begin();
+        mutation.replace_node_discard_trivia(
+            state.block.items(),
+            state
+                .sorted_properties
+                .as_sorted(&state.original_properties)?,
+        );
+
+        return Some(RuleAction::<CssLanguage>::new(
+            ActionCategory::QuickFix(Cow::Borrowed("")),
+            ctx.metadata().applicability(),
+            markup! { "Sort these properties" }.to_owned(),
+            mutation,
+        ));
```

Comment:
```suggestion
        return Some(CssRuleAction::new(
            ActionCategory::QuickFix(Cow::Borrowed("")),
            ctx.metadata().applicability(),
            markup! { "Sort these properties" }.to_owned(),
            mutation,
        ));
```

You probably missed some information from out [contribution guide](https://github.com/biomejs/biome/blob/eedb22edf7ec27d847ca31d0c101f8fa5b4956fd/crates/biome_analyze/CONTRIBUTING.md?plain=1#L244-L258). Can you please update the code accordingly? The first argument is wrong